paddle/platform/assert.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#define STRINGIFY(x) #x
+#define TOSTRING(x) STRINGIFY(x)
+
+#if defined(__APPLE__) && defined(__CUDA_ARCH__) && !defined(NDEBUG)
+#include <stdio.h>
+#define PADDLE_ASSERT(e)                                           \
+  do {                                                             \
+    if (!(e)) {                                                    \
+      printf("%s:%d Assertion `%s` failed.\n", __FILE__, __LINE__, \
+             TOSTRING(e));                                         \
+      asm("trap;");                                                \
```

Comment:
`asm("trap")` is inconsistent in Majel. Because I found in `abort.h`, gdiamos implemented 

```c++
__device__ inline void abort_gpu_kernel(const char* message) {
    printf("Aborted GPU Kernel: `%s` .\n", message);
    #if !defined(__APPLE__)
    asm("trap;");
    #endif
}
```

However, this PADDLE_ASSERT macro only works when it runs on APPLE.

I guess maybe it's 

```
!defined(__APPLE__) && defined(__CUDA_ARCH__) && !defined(NDEBUG)
```